pkg/virt-controller/watch/migration_test.go

```diff
@@ -1586,6 +1597,124 @@ var _ = Describe("Migration watcher", func() {
 		})
 
 	})
+
+	Context("Migration monitoring", func() {
+		var (
+			vmi           *virtv1.VirtualMachineInstance
+			migration     *virtv1.VirtualMachineInstanceMigration
+			targetPod     *k8sv1.Pod
+			attachmentPod *k8sv1.Pod
+			labels        prometheus.Labels
+		)
+
+		BeforeEach(func() {
+			vmi = newVirtualMachineWithHotplugVolume("testvmi-monitoring-"+rand.String(16), virtv1.Running)
+			migration = newMigration("testmigration", vmi.Name, virtv1.MigrationPending)
+			targetPod = newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+			attachmentPod = newAttachmentPodForVirtualMachine(targetPod, migration, k8sv1.PodRunning)
+
+			vmi.Status.NodeName = "node02"
+			targetPod.Spec.NodeName = "node01"
+			labels = prometheus.Labels{"vmi": vmi.Name, "source": vmi.Status.NodeName, "target": targetPod.Spec.NodeName}
+		})
+
+		It("should update metrics on pending migration", func() {
+			pending := prometheustestutil.ToFloat64(migrations.CurrentPendingMigrations.With(labels))
```

Comment:
to make these sort of calls more clear you can use a helper function:
```go
getGaugeCount := func(gauge *GaugeVec) float64 {
    prometheustestutil.ToFloat64(gauge.With(labels))
}
```

Reply:
updated